### PR TITLE
Assignment1

### PR DESCRIPTION
**Pull request by Veltoc**
Don't know if i did enough unit tests since you there is a laundry list of tests that can be done and it wasn't specified... for now I stuck to what was shown to us. If more is needed that would be a nice comment to have. As well as if I did .props and .targets correct.

### CODE REVIEW

**Comment by kenny2892** [comment]
I will review this.

File: SecretSanta/Tests/SecretSanta.Business.Tests/UserTests.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+            //const User user;
+
+            //act
+            User user = new User(id, firstName, lastName, gifts);
+            //assert 
+            Assert.AreEqual<int>(id, user.Id, "Id value is unexpected");
+            Assert.AreEqual<string>(firstName, user.FirstName, "FirstName value is unexpected");
+            Assert.AreEqual<string>(lastName, user.LastName, "LastName value is unexpected");
+            Assert.AreEqual<List<Gift>>(gifts, user.Gifts, "Gifts value is unexpected");
```
Comment (kenny2892) [inline SecretSanta/Tests/SecretSanta.Business.Tests/UserTests.cs]: You don't need the "<>" after "Assert.AreEqual" if the objects being compared are of the same class.

**Comment by Veltoc** [comment]
DevOps: https://dev.azure.com/Masonguer/EWU379-2020/_build?definitionId=2&_a=summary

File: SecretSanta/Tests/SecretSanta.Business.Tests/GiftTests.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+            Assert.AreEqual<string>(title, gift.Title, "Title value is unexpected");
+            Assert.AreEqual<string>(description, gift.Description, "Description value is unexpected");
+            Assert.AreEqual<string>(url, gift.Url, "Url value is unexpected");
+            Assert.AreEqual<User>(user, gift.User, "User is unexpected");
+        }
+        [TestMethod]
+        [ExpectedException(typeof(ArgumentNullException))]
+        public void Create_VerifyPropertiesAreNotNull_NotNull()
+        {
+            Gift gift = new Gift(0, "","","", new User(0, "02", "03", new List<Gift>()));
```
Comment (Keboo) [inline SecretSanta/Tests/SecretSanta.Business.Tests/GiftTests.cs]: In cases like this you can use a throw away variable to tell the compiler to not allocate space for it.
```suggestion
            var _ = new Gift(0, "","","", new User(0, "02", "03", new List<Gift>()));
```

File: SecretSanta/src/SecretSanta.Business/Gift.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        public string Description
+        {
+            get => _Description;
+            set => _Description = string.IsNullOrWhiteSpace(value) ? throw new ArgumentNullException(nameof(value)):value;
+        }
+        private string _Url = "<Invalid>";
+        public string Url
+        {
+            get => _Url;
+            set => _Url = string.IsNullOrWhiteSpace(value) ? throw new ArgumentNullException(nameof(value)) : value;
```
[the remark this answers is not in the document]
Comment (Keboo) [reply SecretSanta/src/SecretSanta.Business/Gift.cs]: Agreed. Though you will find there are cases in .NET that also behave like this.

**Comment by Veltoc** [comment]
https://dev.azure.com/Masonguer/EWU379-2020/_build/results?buildId=8&view=results
it ran on a pull request on assignment 1, also what did i miss on the nullability? something mentioned in class was he was telling us to add nullability to 1 specifically then later to it all so i assumed the conditional was just for API, and the .props has nullability(conditional wasn't specified)

**Pull request closed without merging**